Lib/ipaddress.py

```diff
@@ -2180,34 +2136,18 @@ def __init__(self, address, strict=True):
               an IPv6 address.
             ValueError: If strict was True and a network address was not
               supplied.
-
         """
-        _BaseNetwork.__init__(self, address)
-
-        # Constructing from a packed address or integer
-        if isinstance(address, (int, bytes)):
-            addr = address
-            mask = self._max_prefixlen
-        # Constructing from a tuple (addr, [mask])
-        elif isinstance(address, tuple):
-            addr = address[0]
-            mask = address[1] if len(address) > 1 else self._max_prefixlen
-        # Assume input argument to be string or any object representation
-        # which converts into a formatted IP prefix string.
-        else:
-            args = _split_optional_netmask(address)
-            addr = self._ip_int_from_string(args[0])
-            mask = args[1] if len(args) == 2 else self._max_prefixlen
+        super().__init__(address)
```

Comment:
What is faster, `_BaseNetwork.__init__()` or `super().__init__()`?

Reply:
In my initial patch I changed this to the `super().__init__()` on purpose from the version where the base class is mentioned explicitly. Not using the `super()` version here makes it harder to write subclasses for those classes. Also, I personally think it's better style to use `super()`.

I have not measured it but I don't think that this is a significant speed improvement.

Reply:
> Not using the `super()` version here makes it harder to write subclasses for those classes.

Simple subclass is OK without super().  It helps only complicated subclassing:

```py
class A(IPv4Network):
    def __init__(self, address):
        super().__init__(address)

class B(_BaseNetwork):
    def __init__(self, address)
        super().__init__(address)

class C(A, B):
    def __init__(self, address):
        super().__init__(address)
```

In this case, if we don't use `super()`, C -> A -> IPv4Network -> _BaseNetwork (SubB is skipped)
if we use `super()`, C -> A -> IPv4Network -> B -> _BaseNetwork.

But note that `_BaseNetwork` is private class.
Not using `super()` here doesn't make write subclass of IPv4Network harder.

> Also, I personally think it's better style to use `super()`.

I'm +0.1 on super in general.

> I have not measured it but I don't think that this is a significant speed improvement.

It seems 3% slowdown.

```
$ ./python -m perf timeit -s 'from ipaddress import IPv4Network' 'IPv4Network("10.10.1.0/24")'
...
Mean +- std dev: [legacy] 8.96 us +- 0.09 us -> [super] 9.23 us +- 0.07 us: 1.03x slower (+3%)
```

I don't have any strong opinion here.

Reply:
Now there is no `__init__` call.